### ISSUE TITLE
UInt instances?

### ISSUE THREAD

**Issue by jamesdbrock** [purescript-contrib/purescript-uint#17]
Should we add some of these instances?

https://pursuit.purescript.org/packages/purescript-uint-instances/

**Comment by maxdeviant** [comment]
Looks like there was some previous discussion in #7 on this.

From what I can tell, the outcome of that issue was the addition of an `Arbitrary` instance in d3a6424ca266d5c8e2388e8b81bf4195f9a0e303.

The `Arbitrary` instance was then removed in 6820b50a97c4ae993b6ed34dede182063cdf97e4 when the QuickCheck dependencies were moved into the test dependencies.

Looking at the `purescript-uint-instances` package, it adds the following instances that we do not currently have:

- [`Arbitrary`](https://pursuit.purescript.org/packages/purescript-quickcheck/7.1.0/docs/Test.QuickCheck.Arbitrary#t:Arbitrary) and [`Coarbitrary`](https://pursuit.purescript.org/packages/purescript-quickcheck/7.1.0/docs/Test.QuickCheck.Arbitrary#t:Coarbitrary) from `purescript-quickcheck`
- [`EncodeJson`](https://pursuit.purescript.org/packages/purescript-argonaut-codecs/8.1.0/docs/Data.Argonaut.Encode.Class#t:EncodeJson) and [`DecodeJson`](https://pursuit.purescript.org/packages/purescript-argonaut-codecs/8.1.0/docs/Data.Argonaut.Decode.Class#t:DecodeJson) from `purescript-argonaut-codecs`
- [`Hashable`](https://pursuit.purescript.org/packages/purescript-unordered-collections/2.1.4/docs/Data.Hashable#t:Hashable) from `purescript-unordered-collections`

**Comment by chtenb** [comment]
I came here because I was looking for an Arbitrary instance for UInt. Is there a reason why we would not want to expose the Arbitrary instance in the API?

**Comment by garyb** [comment]
I think providing something in terms of `MonadGen` would be preferable since that has a much lower dependency overhead.

**Comment by jamesdbrock** [comment]
I would love a MonadGen PR @chtenb ... or Arbitrary, or whatever you think is best

**Comment by chtenb** [comment]
Right, so if `MonadGen` is preferred then we would have something like

```hs
chooseUInt :: forall gen. MonadGen gen => UInt -> UInt -> gen UInt
chooseUInt a b = fromInt $ chooseInt (toInt a) (toInt b)
```

but we're using the `chooseInt` from quickCheck here, so that doesn't solve the dependency overhead.

Wouldn't it perhaps be better to add this to the quickCheck library instead then? The quickcheck lib would gain uint as a dependency, but that's a very small library. Another advantage would be that we could add Arbitrary while we're at it.

**Comment by jamesdbrock** [comment]
You don't need the quickcheck dependency, you can just use the `chooseInt` provided by the `MonadGen`. See for example https://pursuit.purescript.org/packages/purescript-float32/2.0.0/docs/Data.Float32.Gen

**Comment by chtenb** [comment]
You're right @jamesdbrock 
So in that case I'd say we add the chooseUInt based on MonadGen to this package and then also provide the arbitrary instance in the quickcheck package, if the folks there are okay with that.

**Comment by chtenb** [comment]
We seem to have overlooked that genUInt already exists in `UInt.Gen` 😄
